packages/react-components/src/lib/context/AgoricProviderLite.tsx

```diff
@@ -0,0 +1,219 @@
+        console.error('got watchPublicSubscribers err', err);
+      });
+    }
+
+    return () => {
+      isCancelled = true;
+      setPurses(undefined);
+      setIsSmartWalletProvisioned(undefined);
+      setOfferIdsToPublicSubscribers(undefined);
+    };
+  }, [walletConnection]);
+
+  useEffect(() => {
+    const getWatcher = async () => {
+      const restEndpoint = await chain.getRestEndpoint(useCustomEndpoints);
+      const apiAddr =
+        typeof restEndpoint === 'string' ? restEndpoint : restEndpoint.url;
+      const watcher = makeAgoricChainStorageWatcher(
+        apiAddr,
+        chainName,
+        onConnectionError
+      );
+      setChainStorageWatcher(watcher);
+    };
+
+    getWatcher().catch((e) => {
+      console.error('error making agoric chain storage watcher', e);
+      onConnectionError(e);
+    });
+  }, []);
+
+  useEffect(() => {
+    const getAgoricWalletConnection = async () => {
+      const [rpcEndpoint, aminoSigner] = await Promise.all([
+        chain.getRpcEndpoint(useCustomEndpoints),
+        client?.getOfflineSigner?.(chain.chain.chain_id, 'amino'),
+      ]);
+      const rpcAddr =
+        typeof rpcEndpoint === 'string' ? rpcEndpoint : rpcEndpoint.url;
+
+      assert(
+        aminoSigner !== undefined,
+        'Cannot getOfflineSigner, is wallet connected?'
+      );
+      const signingStargateClient =
+        await SigningStargateClient.connectWithSigner(
+          rpcEndpoint,
+          // @ts-expect-error Some mismatch on accountNumber 'Long' types
+          aminoSigner,
+          {
+            aminoTypes: new AminoTypes({
+              ...agoricConverters,
+              ...createBankAminoConverters(),
+              ...createAuthzAminoConverters(),
+            }),
+            registry: new Registry([
+              ...defaultRegistryTypes,
+              ...agoricRegistryTypes,
+            ]),
+          }
+        );
+      assert(chain.address, 'Cannot get address, is wallet connected?');
+      assert(chainStorageWatcher);
+      const agoricWalletConnection = await makeAgoricWalletConnection(
+        chainStorageWatcher,
+        rpcAddr,
+        (e: unknown) => {
+          console.error('wallet connection error', e);
+          onConnectionError(e);
+        },
+        { address: chain.address, client: signingStargateClient }
+      );
+      setWalletConnection(agoricWalletConnection);
+    };
+
+    if (
+      status === 'Done' &&
+      chain.address !== undefined &&
+      chainStorageWatcher !== undefined
+    ) {
+      getAgoricWalletConnection().catch((e) => {
+        onConnectionError(e);
+        console.error('error making agoric wallet connection', e);
+      });
+    }
+
+    return () => {
+      setWalletConnection(undefined);
+    };
+  }, [status, chain.address, chainStorageWatcher]);
+
+  const state = {
+    address: chain.address,
+    chainName,
+    connect: chain.connect,
+    chainStorageWatcher,
+    walletConnection,
```

Comment:
Should we maybe just expose `makeOffer` and `provisionSmartWallet` here? What else would a consumer need from walletConnection?

Reply:
Yea I was being conservative about what I exposed in my initial pass, but those seem like worthy candidates now that you mention it. Done.